mllib/src/main/scala/org/apache/spark/ml/nlp/CRF.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.nlp
+
+import scala.collection.mutable.ArrayBuffer
+import org.apache.spark.annotation.DeveloperApi
+
+private[spark] class CRF {
+  private val freq: Integer = 1
+  private val maxiter: Integer = 100000
+  private val cost: Double = 1.0
+  private val eta: Double = 0.0001
+  private val C: Float = 1
+  private val threadNum: Integer = Runtime.getRuntime.availableProcessors()
+  private val threadPool: Array[CRFThread] = new Array[CRFThread](threadNum)
```

Comment:
It might be better to support distribution with Spark API not local thread. Such as transformation to RDD and partitions.